python/MDSplus/connection.py

```diff
@@ -175,39 +182,80 @@ def __getAnswer__(self, to_msec=-1):
             if mem.value is not None:
                 _MdsIpFree(mem)
 
+    def _send_arg(self, value, idx, num):
+        """Internal routine to send argument to mdsip server"""
+        val = _dat.Data(value)
+        if not isinstance(val, _sca.Scalar) and not isinstance(val, _arr.Array):
+            val = _dat.Data(val.data())
+        info = self._inspect(val)
+        _exc.checkStatus(
+            _SendArg(self.conid,
+                     idx,
+                     info['dtype'],
+                     num,
+                     info['length'],
+                     info['dimct'],
+                     info['dims'].ctypes.data,
+                     info['address']))
+
+    def get(self, exp, *args, **kwargs):
+        if 'arglist' in kwargs:
+            args = kwargs['arglist']
+        timeout = kwargs.get('timeout', -1)
+        num = len(args)+1
+        exp = _ver.tobytes(exp)
+        _exc.checkStatus(_SendArg(self.conid, 0, 14, num,
+                                  len(exp), 0, 0, ctypes.c_char_p(exp)))
+        for i, arg in enumerate(args):
+            self._send_arg(arg, i+1, num)
+        return self._get_answer(timeout)
+
+
+class Connection(object):
+    """Implements an MDSip connection to an MDSplus server"""
+
+    @property
+    def local(self):
+        try:
+            local = self._local.local
```

Comment:
We could consider calling this `self._local.conn` or something similar

Reply:
like so ?